examples/creatives/hover_nonlinear.js

```diff
@@ -0,0 +1,143 @@
+const QUARTER_THE_SIZE = .25;
+const THIRTY_FIVE_PERCENT = .35;
+const HALF_THE_SIZE = .5;
+const THREE_QUARTERS_THE_SIZE = .75;
+
+/* This creative expands and collapses when the user hovers over the banner. */
+class HoverNonLinear extends BaseSimidCreative {
+    constructor() {
+        super();
+
+        this.initialDimensions_ = null;
+    }
+
+    /** @override */
+    onInit(eventData) {
+        super.onInit(eventData);
+        this.initialDimensions_ = this.environmentData.creativeDimensions;
+        this.videoDimensionHeight = this.environmentData.videoDimensions.height;
+        this.videoDimensionWidth = this.environmentData.videoDimensions.width;
+    }
+
+    /** @override */
+    onStart(eventData) {
+        super.onStart(eventData);
+        this.addActions_();
+    }
+
+    /**
+     * Adds actions to different buttons available on the overlay.
+     * @private
+     */
+    addActions_() {
+        this.sendMessageOnClick_('close_ad', CreativeMessage.REQUEST_STOP);
+        this.onHover_('content_container', 'mouseover');
+        this.onMouseOut_('content_container', 'mouseout');
+    }
+
+    /**
+     * Sends a SIMID message whenever an element is clicked.
+     * @param {String} elementName The name of the element.
+     * @param {String} message The message to send to the player.
+     * @private
+     */
+    sendMessageOnClick_(elementName, message) {
+        const sendMessageFunction = () => {this.simidProtocol.sendMessage(message);}
+        document.getElementById(elementName).addEventListener('click', sendMessageFunction);
+    }
+
+    /**
+     * Adds a hover event listener to the contents of the iframe that expands the iframe.
+     * @param {String} elementName The name of the element.
+     * @param {Event} event The event performed on the element.
+     * @private
+     */   
+    onHover_(elementName, event) {
+        const expandOnHoverFunction = () => {
+            const newDimensions = {};
+            let resizeParams = {};
+
+            //if ad is at the top quarter of the player
+            if (this.initialDimensions_.y < (this.videoDimensionHeight * QUARTER_THE_SIZE)) {
+                newDimensions.height = (this.videoDimensionHeight * THIRTY_FIVE_PERCENT);
+                newDimensions.x = 0;
+                newDimensions.y = this.initialDimensions_.y;
+                newDimensions.width = this.videoDimensionWidth;
+
+                resizeParams = {
+                    creativeDimensions: newDimensions,
+                    videoDimensions: this.environmentData.videoDimensions,
+                };
+            
+                this.requestResize(resizeParams);
```

Comment:
this whole chunk of code could be shared across the conditionals as well and should be put after the last if statement

Reply:
totally restructured code